rules/CodingStyle/ClassNameImport/UseImportsTraverser.php

```diff
@@ -74,8 +74,11 @@ private function traverseForType(array $stmts, callable $callable, int $desiredT
 
                     $callable($useUse, $name);
                 }
+
+                return null;
```

Comment:
since our logic here already traverses into such nodes, the traverser itself doesn't need todo it a second time

Reply:
@staabm I am testing it, it cause overlapped type 

```diff
diff --git a/rules/CodingStyle/ClassNameImport/ClassNameImportSkipper.php b/rules/CodingStyle/ClassNameImport/ClassNameImportSkipper.php
index 7e3b49fa0f..25e5d6c605 100644
--- a/rules/CodingStyle/ClassNameImport/ClassNameImportSkipper.php
+++ b/rules/CodingStyle/ClassNameImport/ClassNameImportSkipper.php
@@ -67,7 +67,7 @@ final class ClassNameImportSkipper
     public function isFoundInUse(Name $name, array $uses): bool
     {
         $stringName = $name->toString();
-        $nameLastName = strtolower($name->getLast());
+        $nameLastName = strtolower((string) $name->getLast());
```

when changing import type :), so return null seems needed, I will. create separate PR :)

Reply:
double run seems resolve it:

```diff
1) rules/CodingStyle/ClassNameImport/ClassNameImportSkipper.php:66

    ---------- begin diff ----------
@@ @@
     public function isFoundInUse(Name $name, array $uses): bool
     {
         $stringName = $name->toString();
-        $nameLastName = strtolower((string) $name->getLast());
+        $nameLastName = strtolower($name->getLast());

         foreach ($uses as $use) {
             $prefix = $this->useImportsResolver->resolvePrefix($use);
    ----------- end diff -----------

Applied rules:
 * RecastingRemovalRector
```

I guess it is a glitch somewhere, actually, return null is fine, as the `Use_` and `GroupUse` actually has limited attributes.

Reply:
would a `return null` resolve these problems?

Reply:
It doesn't 😁

It probably another problem with detecting type as mixed/errortype when docblock changed, probably related with @TomasVotruba change on:

- https://github.com/rectorphp/rector-src/pull/4986

which type now no longer immediatelly got mixed when docblock changed

Reply:
@staabm I created new issue for it:

- https://github.com/rectorphp/rector/issues/8196

it may wait for @TomasVotruba work on : 

- https://github.com/rectorphp/rector-src/pull/4999

for update docblock content right in the rule, which type currently goes mixed.

Reply:
Thanks 🙏

Reply:
@staabm resolved by PR:

- https://github.com/rectorphp/rector-src/pull/4999

Ref https://getrector.com/demo/ead4b18f-d042-456d-bcdb-3caef53cc16f